docs/developing_recipes_locally.md

```diff
@@ -0,0 +1,201 @@
+
+    # Lint the test_package
+    pylint --rcfile=linter/pylintrc_testpackage recipes/boost/all/test_package/conanfile.py
+    ```
+
+## Testing the different `test_*_package`
+
+This can be selected when calling `conan create` or sperately with `conan test`
+
+```sh
+conan create recipes/fmt/all/conanfile.py 9.0.0@ -tf test_v1_package/
+```
+
+```sh
+conan test recipes/fmt/all/test_v1_package/conanfile.py fmt/9.0.0@
+```
+
+## Testing more environments
+
+This can be difficult for some platforms given virtualization support.
+
+For Windows and MacOS users, you can test the Linux build environments by with the the Docker build images.
+
+Assuming you've already tested it locally, and it's successfully export to your cache, you can:
+
+1. Creating a new profile.
+2. Build missing packages
+
+Example.
+
+```
+docker run -v/Users/barbarian/.conan:/home/conan/.conan conanio/gcc8 bash -c "conan profile new --detect gcc8"
+docker run -v/Users/barbarian/.conan:/home/conan/.conan conanio/gcc8 bash -c "conan install -pr gcc8 fmt/9.0.0@ --build missing"
+```
+
+## Using Conan 2.0
+
+Everything you need to know about the methods, commands line, outputs can be found in the
+[Conan 2.0 Migrations](https://docs.conan.io/en/latest/conan_v2.html) docs. This should be fairly straight forward.
+Conan 2.0 by default has a different `CONANN_USER_HOME` location. You can have to separate caches, profiles
+and settings. So no need to worry about breaking your v1 setup.
+
+> **Note**: There are pretty heavy changes to the CLI so very few of the commands will have survived.
+> [Unified Command Pattern](https://docs.conan.io/en/latest/migrating_to_2.0/commands.html#unified-patterns-in-command-arguments)
+> for example changes how settings and options are passed.
+
+
+Simply install Conan with `pip install conan --upgrade --pre`
+
+You can confirm the installation with
+
+```sh
+$ conan --version
+Conan version 2.0.0-beta2
+$ conan config home
+Current Conan home: /Users/barbarian/.conan2
+```
+
+> **Note**: You will most likely see
+> ```
+> Initialized file: '/Users/christopherm/.conan2/settings.yml'
+> Initialized file: '/Users/christopherm/.conan2/extensions/plugins/compatibility/compatibility.py'
+> Initialized file: '/Users/christopherm/.conan2/extensions/plugins/compatibility/app_compat.py'
+> Initialized file: '/Users/christopherm/.conan2/extensions/plugins/compatibility/cppstd_compat.py'
+> Initialized file: '/Users/christopherm/.conan2/extensions/plugins/profile.py'
+> ```
+> When running the client for the first time.
+
+You will need to setup profiles
+
+```sh
+conan profile detect
+```
+
+> **Warning**: This is a best guess, you need to make sure it's correct.
+
+There is a remote for the 2.0 migration binaries which can be added
+
+```
+conan remote add conanv2 https://conanv2beta.jfrog.io/artifactory/api/conan/conan --index 0
+```
+
+Trying build an existing recipe
+
+```sh
+cd recipes/fmt
+conan create all/conanfile.py --version 9.0.0
```

Comment:
Is not dual profile mandatory for Conan 2.0?

Reply:
It's the defaults that changed. But you do not need to manually specify it

the defaults match what the 1.x docs are showing